api/client/src/main/java/org/projectnessie/client/NessieConfigConstants.java

```diff
@@ -44,6 +44,44 @@ public final class NessieConfigConstants {
    * authentication.
    */
   public static final String CONF_NESSIE_AUTH_TOKEN = "nessie.authentication.token";
+
+  /**
+   * Required. The URL of the OAuth2 token endpoint, including the path to the token endpoint when
+   * using OAuth2 authentication. For Keycloak, this is typically {@code
+   * https://<keycloak-server>/auth/realms/<realm-name>/protocol/openid-connect/token}.
+   */
+  public static final String CONF_NESSIE_OAUTH_TOKEN_ENDPOINT =
+      "nessie.authentication.oauth.token-endpoint";
+
+  /** Required. The client ID to use when using OAuth2 authentication. */
+  public static final String CONF_NESSIE_OAUTH_CLIENT_ID = "nessie.authentication.oauth.client-id";
+
+  /** Required. The client secret to use when using OAuth2 authentication. */
+  public static final String CONF_NESSIE_OAUTH_CLIENT_SECRET =
+      "nessie.authentication.oauth.client-secret";
+
+  /**
+   * The default token validity when using OAuth2 authentication. Defaults to PT1H (one hour). Must
```

Comment:
*sigh* I suspected, that the standard(s) require that an expiration time is sent with the token - but it's only recommended :(